src/main/java/org/pac4j/dropwizard/J2EHelper.java

```diff
@@ -0,0 +1,161 @@
+
+        registerFilter(environment, config, SecurityFilter.class,
+                fConf.getMapping(), initParameters);
+    }
+
+    public static void registerCallbackFilter(Environment environment,
+            Config config, ServletCallbackFilterConfiguration fConf) {
+
+        final Map<String, String> initParameters = new HashMap<>();
+
+        set(initParameters, Pac4jConstants.DEFAULT_URL, fConf.getDefaultUrl());
+        set(initParameters, Pac4jConstants.MULTI_PROFILE,
+                fConf.getMultiProfile());
+        set(initParameters, Pac4jConstants.RENEW_SESSION,
+                fConf.getRenewSession());
+
+        registerFilter(environment, config, CallbackFilter.class,
+                fConf.getMapping(), initParameters);
+    }
+
+    public static void registerLogoutFilter(Environment environment,
+            Config config, ServletLogoutFilterConfiguration fConf) {
+
+        final Map<String, String> initParameters = new HashMap<>();
+
+        set(initParameters, Pac4jConstants.DEFAULT_URL, fConf.getDefaultUrl());
+        set(initParameters, Pac4jConstants.LOGOUT_URL_PATTERN,
+                fConf.getLogoutUrlPattern());
+
+        registerFilter(environment, config, ApplicationLogoutFilter.class,
+                fConf.getMapping(), initParameters);
+    }
+
+    private static void set(Map<String, String> initParameters, String key,
+            Object o) {
+        if (o != null) {
+            initParameters.put(key, o.toString());
+        }
+    }
+
+    private static void registerFilter(Environment environment, Config config,
+            Class<? extends Filter> clazz, String mapping,
+            Map<String, String> initParameters) {
+
+        final FilterRegistration.Dynamic filterRegistration = environment
+                .servlets().addFilter(clazz.getName(), clazz);
+
+        // let's create a ConfigFactory implementation unique to this filter
+        String ccName = createUniqueConfigFactory(config,
+                clazz.getSimpleName() + filterRegistration.hashCode());
+
+        set(initParameters, Pac4jConstants.CONFIG_FACTORY, ccName);
+
+        filterRegistration.addMappingForUrlPatterns(
+                EnumSet.of(DispatcherType.REQUEST), true, mapping);
+
+        filterRegistration.setInitParameters(initParameters);
+    }
+
+    /**
+     * We need a {@link ConfigFactory} for the {@link Config} for each J2E
+     * {@link Filter}s. We could exploit {@link ConfigSingleton} but in order to
+     * avoid conflict with other uses of it, we prefer to have our own. And even
+     * though, for now, there seems to be only one {@link Config} for the whole
+     * application, it is cleaner to generate one {@link ConfigFactory} per
+     * filter instead of using always the same one: who knows how things will
+     * evolve in the future?
+     * 
+     * @author Victor Noel - Linagora
+     *
+     */
+    private static String createUniqueConfigFactory(Config config,
```

Comment:
Let me reformulate: you already have a `Config` object, why not build a `DefaultConfigFactory`  with it (from the `pac4j-core` module: https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/main/java/org/pac4j/core/config/DefaultConfigFactory.java) to have the config factory  return this `Config` object?